File: app/javascript/channels/operation_channel.js
import consumer from "./consumer"

$(function() {
	const operationChannel = consumer.subscriptions.create({ channel: "OperationChannel", data_logger_id: $(".data_logger").attr("data_logger_id") }, {
		operate(operation) {
			return this.perform('operate', {
				"operation": operation
			});
		}
	});

	$('#operation-button').on('click', function(){
		operationChannel.operate($('#operation-input').val());
	});
});

File: app/channels/operation_channel.rb
class OperationChannel < ApplicationCable::Channel
  def subscribed
    stream_from channel_name
  end

  def unsubscribed
  end

  def operate(data)
    ActionCable.server.broadcast channel_name, body: data["operation"]
  end

  private
    def channel_name
      "operation_channel"
      #"operation_channel:#{params[:data_logger_id]}"
    end
end
